File: src/main/java/com/chattriggers/ctjs/minecraft/wrappers/objects/threading/ThreadHandler.java
package com.chattriggers.ctjs.minecraft.wrappers.objects.threading;

import java.util.ArrayList;
import java.util.List;

public class ThreadHandler {
    private List<WrappedThread> activeThreads;

    public ThreadHandler() {
        this.activeThreads = new ArrayList<>();
    }

    public void addThread(WrappedThread thread) {
        this.activeThreads.add(thread);
    }

    public void removeThread(WrappedThread thread) {
        activeThreads.removeIf(thread1 -> thread1 == thread);
    }

    public void stopThreads() {
        activeThreads.forEach(Thread::interrupt);

        activeThreads.clear();
    }
}

File: src/main/kotlin/com/chattriggers/ctjs/minecraft/wrappers/objects/threading/ThreadHandler.kt
package com.chattriggers.ctjs.minecraft.wrappers.objects.threading

import java.util.*
import java.util.function.Consumer

class ThreadHandler {
    private val activeThreads: MutableList<WrappedThread>

    init {
        this.activeThreads = ArrayList()
    }

    fun addThread(thread: WrappedThread) =
        this.activeThreads.add(thread)

    fun removeThread(thread: WrappedThread) =
        activeThreads.removeIf { thread1 -> thread1 === thread }

    fun stopThreads() {
        activeThreads.forEach(Consumer<WrappedThread> { it.interrupt() })
        activeThreads.clear()
    }
}